niva.md

```diff
@@ -0,0 +1,319 @@
+---
+name: niva
+filename: main.niva
+contributors:
+    - ["gavr", "https://github.com/gavr123456789"]
+---
+
+## Intro
+Niva is a simple language that takes a lot of inspiration from Smalltalk.  
+But leaning towards the functional side and static typed.
+Everything is still an object, but instead of classes, interfaces, inheritance, and abstract classes,  
+we have tagged unions, which is the only way to achieve polymorphism.
+
+
+For example, everything except the declaration is sending messages to objects.  
+`1 + 2` is not a + operator, but a `... + Int` message for `Int` object.
+(ofc there are no extra costs for that)   
```

Comment:
This is first and foremost documentation, which probably should avoid `ofc` etc.